### DISCUSSION TITLE
If there are multiple Secondary storage, how does CS decide where to place the next snapshot - related to issue

### DISCUSSION THREAD

**Discussion by nxsbi** [apache/cloudstack#9431] [Q&A]
I created an issue https://github.com/apache/cloudstack/issues/9430 and had some questions around the behind the scene process.

QUick summary - I had 1 secondary storage. I added second one, and moved (blanaced) snapshots to second storage. Looked like it all worked. However, the next day onwards, what i observed is that no new snapshots were taking palce for the volumes that went to the new secondary storage. 

Upon inspection of logs, I noticed that the command is trying to mount the folder on the original storage location (which does not exist anymore because it moved to new store). 

I looked at the snapshot_store_ref table, and see that all snapshots for the specific volume_id are pointing to the new secondary store. 

How is it that CS is trying to mount the existing (old) server path? 

Besides the snapshot_store_ref, is there some other palce that CS is looking - it seems it did still thinks the snapshots live on old path.

**Comment by DaanHoogland** [comment]
@nxsbi , did you find an answer to this query?

**Reply by nxsbi** [reply]
Not really. I eventually moved to a single secondary storage so didn’t have this happening again. 

However for closing the topic if you would provide explanation, that would be great.

**Reply by DaanHoogland** [reply]
I don’t think I have an answer just now. Let’s keep it open until someone has.

**Comment by DaanHoogland** [comment]
@nxsbi , this is getting dusty. By the looks of it @Pearl1594 asked a question on #9430 . Is it still worth going into this?

**Comment by DaanHoogland** [comment]
@nxsbi , closing this, please follow up with more information if you feel this is still an issue.
